### ISSUE TITLE
dotnet restore fails on Windows with .NET Core 2.0-preview1-005977

### ISSUE THREAD

**Issue by livarcocc** [NuGet/Home#5386]
_From @martincostello on June 11, 2017 14:5_

## Steps to reproduce

Clone the `DotNet-Core-2.0` branch of https://github.com/martincostello/website.git (see martincostello/website#194).

## Expected  behavior

`dotnet restore` completes successfully, after which the build and tests succeed.

## Actual behavior

On Windows `dotnet restore` fails (see AppVeyor build [ere](https://ci.appveyor.com/project/martincostello/website/build/1.0.365#L108) and [here](https://ci.appveyor.com/project/martincostello/website/build/1.0.365#L340)):

```ps
C:\projects\website\.dotnetcli\sdk\2.0.0-preview1-005977\NuGet.targets(97,5): error : Unable to resolve 'C:\projects\website\src\Website\Website.csproj' for '.NETCoreApp,Version=v2.0'. [C:\projects\website\Website.sln]

Errors in C:\projects\website\tests\Website.Tests\Website.Tests.csproj
    Unable to resolve 'C:\projects\website\src\Website\Website.csproj' for '.NETCoreApp,Version=v2.0'.
```

On Linux (see [Travis build](https://travis-ci.org/martincostello/website/builds/241732947)), the build succeeds as expected.

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.0.0-preview1-005977)

Product Information:
 Version:            2.0.0-preview1-005977
 Commit SHA-1 hash:  414cab8a0b

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Coding\website\.dotnetcli\sdk\2.0.0-preview1-005977\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview1-002111-00
  Build    : 1ff021936263d492539399688f46fd3827169983
```

_Copied from original issue: dotnet/cli#6828_

**Comment by emgarten** [comment]
This error can occur when the project dependency project is invalid or when the file is missing on disk.

I'm not able repro this locally when restoring. @martincostello are you seeing this locally, or only on appveyor?

**Comment by martincostello** [comment]
I'm seeing it both on my laptop locally and in AppVeyor.

**Comment by emgarten** [comment]
@martincostello can you try with the latest preview2 CLI?  The error messages have been improved in this area, it should give you a slightly more helpful message.

**Comment by martincostello** [comment]
@emgarten Sure, I'll update the tooling later this evening and see what happens.

**Comment by martincostello** [comment]
Issue persists with `2.0.0-preview2-006120`.

**Comment by emgarten** [comment]
@martincostello same error message?

**Comment by martincostello** [comment]
Yep, everything is the same as before. It just says it can't resolve the main project from the test project. There's no further detail compared to with preview 1.

**Comment by martincostello** [comment]
@emgarten This issue appears to have resolved itself with `2.0.0-preview2-006497`.
